evals/src/evaluators/llm_judge.rs

```diff
@@ -0,0 +1,610 @@
+                    .clone()
+            })],
+        }],
+    };
+
+    let params = ClientInferenceParams {
+        function_name: Some(get_llm_judge_function_name(eval_name, evaluator_name)),
+        model_name: None,
+        episode_id: None,
+        input,
+        stream: Some(false),
+        include_original_response: false,
+        params: InferenceParams::default(),
+        variant_name: None,
+        dryrun: Some(false),
+        internal: true,
+        tags: HashMap::from([(
+            "tensorzero::eval_run_id".to_string(),
+            eval_run_id.to_string(),
+        )]),
+        dynamic_tool_params: DynamicToolParams::default(),
+        output_schema: None,
+        credentials: HashMap::new(),
+        cache_options: tensorzero::CacheParamsOptions {
+            max_age_s: None,
+            enabled: CacheEnabledMode::On,
+        },
+    };
+    let result = tensorzero_client.inference(params).await?;
+    let response = match result {
+        InferenceOutput::NonStreaming(response) => response,
+        InferenceOutput::Streaming(..) => {
+            bail!("Streaming not supported for LLM judge evals. This is a bug, please file a bug report at https://github.com/tensorzero/tensorzero/discussions/new?category=bug-reports.")
+        }
+    };
+    let output = match response {
+        InferenceResponse::Chat(..) => {
+            bail!("Chat output not supported for LLM judge evals. This is a bug, please file a bug report at https://github.com/tensorzero/tensorzero/discussions/new?category=bug-reports.")
+        }
+        InferenceResponse::Json(json_response) => json_response
+            .output
+            .parsed
+            .ok_or_else(|| anyhow::anyhow!("JSON output does not contain a `parsed` field"))?,
+    };
+    match llm_judge_config.output_type {
+        LLMJudgeOutputType::Float => Ok(Some(
+            output
+                .get("score")
+                .ok_or_else(|| anyhow::anyhow!("JSON output does not contain a `score` field"))?
+                .clone(),
+        )),
+        LLMJudgeOutputType::Boolean => Ok(Some(
+            output
+                .get("score")
+                .ok_or_else(|| anyhow::anyhow!("JSON output does not contain a `score` field"))?
+                .clone(),
+        )),
+    }
+}
+
+pub fn prepare_serialized_input(resolved_input: &ResolvedInput) -> Result<String> {
+    for message in &resolved_input.messages {
+        for content in &message.content {
+            match content {
+                ResolvedInputMessageContent::Image(..) => {
+                    bail!("Image content not supported for LLM judge evals")
+                }
+                ResolvedInputMessageContent::Unknown { .. } => {
+                    bail!("Unknown content not supported for LLM judge evals")
+                }
+                _ => {}
```

Comment:
Can you explicitly name the other variants here, so that this stops compiling if we add new content types?